src/export/StructureDefinitionExporter.ts

```diff
@@ -173,7 +177,23 @@ export class StructureDefinitionExporter implements Fishable {
             if (rule.path !== '') {
               element.setInstancePropertyByPath(rule.caretPath, rule.value, this);
             } else {
-              structDef.setInstancePropertyByPath(rule.caretPath, rule.value, this);
+              try {
+                structDef.setInstancePropertyByPath(rule.caretPath, rule.value, this);
+              } catch (e) {
+                if (
+                  e instanceof MismatchedTypeError &&
+                  e.valueType === 'string' &&
+                  e.elementType === 'Resource'
```

Comment:
This seems like a safe approach when the user is just setting the `contained` array, but this won't work for something like:
```
* ^identifier = MyIdentifier
```
Which is something to consider. I'm not saying that use case needs to be included in this PR necessarily. But maybe this approach could be refactored so that if we do add that, it is not a huge lift to add. You set the `IsInstance` flag on the caret value rule in the importer. Wouldn't that be a sufficient condition for knowing if a rule should be deferred?

Reply:
That's a good point, and I think that case is within scope for this PR. So maybe I want to keep that `isInstance` flag around, and just check for that? It only gets checked if assigning the string fails, so that shouldn't get me into too much trouble.